### DISCUSSION TITLE
CommentThread Default Expanded doesn't work any more in 1.72.2?

### DISCUSSION THREAD

**Discussion by KrabsWong** [microsoft/vscode-discussions#209] [Extension Development QnA] [answered]
I am developing an extension to get source code from my own server and show up the diff panel with users comments.

All the comments set collapseState to Expanded by the product design. It works well in v1.71, after I upgrade the vscode version to 1.72.2, the default setting doesn't work any more.

The behavior is a little odd. If I don't set anything (use the default value), the comment can be expanded after I click the comment icon. But if I set the default state to Expanded, no comment is expanded by default in the diff editor, after click the comment icon for the first time, nothing happened, and the comment can be expanded after clicking the second time.

```typescript
const newThread = commentController.createCommentThread(
    threadUri,
    new vscode.Range(
      new vscode.Position(lineNumber, 0),
      new vscode.Position(lineNumber, 0),
    ),
    [],
  );
  newThread.label = 'Demo thread label';
  newThread.collapsibleState = vscode.CommentThreadCollapsibleState.Expanded
  const newThreadComments: CustomThreadComment[] = [];
  if (threadComments?.length) {
    newThreadComments.push(...threadComments.reduce((result, commentItem: any) => {
      const { id, body, author, timestamp, resolveState } = commentItem;

      result.push(new CustomThreadComment({
        id,
        body,
        author: {
          name: author.name,
          iconPath: author.iconPath,
        },
        timestamp,
        parent: newThread,
        contextValue,
        resolveState,
      }));
      return result;
    }, []));
  }

  newThread.comments = [...newThread?.comments, ...newThreadComments];
}
```

It seems that there are some new properties added in 1.72.2 for comments, not sure if there is something related about it.

<img width="1380" alt="image" src="https://user-images.githubusercontent.com/12368943/199305335-28c3583e-c51b-4517-b256-31d82ccf6004.png">

BTW, is the default comments setting is "Collapsed" in the diff editor in https://github.dev/ pull request view

**Comment by KrabsWong** [comment, marked as answer]
Codes are OK, something else cause this bug.

**Reply by KrabsWong** [reply]
https://github.com/microsoft/vscode/issues/164471

Track with this issue.

**Reply by isidorn** [reply]
Thanks for following up on that bug, I see it is assigned to November so it should be tackled soon.
